src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java

```diff
@@ -110,23 +116,6 @@ private J.Unary asNegatedUnary(J.MethodInvocation mi) {
                         JavaType.Primitive.Boolean
                 );
             }
-
-            @Override
-            public J visitBinary(J.Binary binary, ExecutionContext ctx) {
-                if (isStringLiteral(binary.getLeft()) || isStringLiteral(binary.getRight())) {
-                    J after = null;
-                    if (binary.getOperator() == J.Binary.Type.Equal) {
-                        after = asEqualsMethodInvocation(binary).withPrefix(binary.getPrefix());
-                    } else if (binary.getOperator() == J.Binary.Type.NotEqual) {
-                        after = asNegatedUnary(asEqualsMethodInvocation(binary)).withPrefix(binary.getPrefix());
-                    }
-                    if (after != null) {
-                        doAfterVisit(new EqualsAvoidsNull().getVisitor());
-                        return after;
-                    }
-                }
-                return super.visitBinary(binary, ctx);
-            }
```

Comment:
Did you check which commit had added this override? That might have missed adding a test. Not sure why it was removed now?

Reply:
I did not remove it, I just moved the method upwards (so public method first, then the private helper methods).

Reply:
I can see it's _old_, but when running with coverage it does appear to be hit.
![image](https://github.com/user-attachments/assets/b6cbaaed-fc05-49fd-bfc8-d2f8efd22ce1)

Reply:
Ah it's merely moved up; that's at times hard to spot in a diff.